src/pages/safe/balances/nfts.tsx

```diff
@@ -39,23 +74,13 @@ const NFTs: NextPage = () => {
           Get the most optimal experience with Safe Apps. View your collections, buy or sell NFTs, and more.
         </Alert>
 
-        {allResults.length > 0 ? (
-          <Nfts collectibles={allResults} />
-        ) : error ? (
-          <ErrorMessage error={error}>Failed to load NFTs</ErrorMessage>
-        ) : (
-          !loading && (
-            <Paper sx={{ py: 9, textAlign: 'center' }}>
-              <Typography variant="h3">No NFTs available</Typography>
-            </Paper>
-          )
-        )}
-
-        {(collectibles?.next || loading) && (
-          <Box py={4}>
-            <LoadMoreButton onLoadMore={() => setPageUrl(collectibles?.next)} loading={loading} />
-          </Box>
-        )}
+        {pages.map((pageUrl, index) => (
+          <NftPage
+            key={index}
+            pageUrl={pageUrl}
+            onNextPage={index === pages.length - 1 ? (pageUrl) => onNextPage(pageUrl) : undefined}
```

Comment:
Why do we need to pass `(pageUrl) => onNextPage(pageUrl)` here instead of just `onNextPage` like for the tx list?